src/main/java/com/squareup/kotlinpoet/ClassName.kt

```diff
@@ -203,6 +213,18 @@ class ClassName private constructor(
       }
       names += getPackage(this).qualifiedName.toString()
       names.reverse()
+      if (names.size == 2 && names[0] == "java.lang") {
+        when (names[1]) {
```

Comment:
Yeah, it felt a bit misplaced and I like an instance method.
That would make it less automatic which is fine as long as it's clearly documented I guess.

I wondered about things like MutableList but it's ambiguous if List should be MutableList or List, perhaps there is a good way to solve that though.